### DISCUSSION TITLE
Feature Request: Overload ATTACH to support file pathing and HTTPFS.

### DISCUSSION THREAD

**Discussion by matsonj** [duckdb/duckdb#6209] [Ideas]
Consider the following scenario:

- Many files in a well-organized folder structure in S3 (or any data lake).
- querying and joining them requires explicit referencing i.e. ```SELECT * FROM '/some/path/*.parquet'```

It would be neat if you could use ```ATTACH /some/path``` to pass in a file path, which would create views for each folder in the file path, creating potentially hundreds of entries in the DuckDB catalog fairly quickly.

So invoking ```show tables;``` would return a list of "tables" aka folders (your file structure would effectively be your data catalog), allowing a clear interface for interacting with your data lake.

This would simplify dev ergonomics: 
- instead of querying the following
  - ```SELECT * FROM '/some/path/entity1/*.parquet entity1 JOIN '/some/path/entity2/.*parquet entity2 ON entity1.key = entity2.key'```
- query is simplified to 
  - ```SELECT * FROM entity1 JOIN entity2 ON entity1.key = entity2.key```

**Comment by Admolly** [comment]
Hi @matsonj .  As a temporary workaround, you could simply create a view corresponding to the file name of each "table" in your folder.

Suppose you have the following structure:

s3://bucket/schema1/table1
s3://bucket/schema1/table2
...
s3://bucket/schema1/table_n

just execute a `CREATE VIEW table_n AS SELECT ...` query for each file and it should be possible to write your queries the way you proposed.

**Comment by matsonj** [comment]
I *think* we have alternative solution landing for this shortly.

https://github.com/duckdb/duckdb/pull/7290
